### DISCUSSION TITLE
AnyStr?

### DISCUSSION THREAD

**Discussion by NicholasSterling** [CarlKCarlK/anyinput#2] [General]
Interesting crate -- thanks for the contribution.

This is a minor point, but when I saw AnyString I thought "oh no, if I had an &str, I wouldn't necessarily want it to be converted to a String."  But the target type for AnyString is actually &str.  So this leaves me with two thoughts:
 1. Would it be clearer to call that AnyStr?
 2. Do you want an AnyString (with a target type of String) too?

**Comment by CarlKCarlK** [comment]
Nicholas,

I very much appreciate your suggestion. (And, I'll likely switch to `AnyStr `in the next version.)

Are you happy with `AnyArray`? Would it be better to call it `AnySlice`?

On the one hand, the names `Str `and `Slice `are what Rust calls things. On the other hand, a newcomer to Rust likely finds `Str` strange and `Slice `confusing.

Aside: In [Python](https://stackabuse.com/python-slice-notation-on-numpy-arrays/) and even in [Rust's ndarray](https://docs.rs/ndarray/latest/ndarray/struct.Slice.html) crate,  a "slice" is a kind of fancy indexing.

Yours,
Carl

**Reply by NicholasSterling** [reply]
Oh, good point.  I actually do think that AnySlice would be better, too.  You're right that Str and Slice may not be familiar terms to those brand new to Rust, but I think in practice by the time people come across your crate and understand why it could be helpful, they'll already be well over the hump with &str and [].  My concern is that somebody who knows some Rust and looks at somebody else's code seeing AnyString and AnyArray is going to think "Ah, I'll bet that is a convenience that gets them a String or array from readily convertible types" and needlessly worry about performance or whatever.

**Reply by Supermath101** [reply]
How about creating a type alias for `AnyStr`, which could internally use `AnySlice`/`AnyArray`? Like this:
```rs
type AnyStr = AnyArray<str>;
```
